src/pip/_internal/cli/cmdoptions.py

```diff
@@ -893,6 +893,16 @@ def check_list_path_option(options):
         )
 
 
+silence_python_deprecation_warnings = partial(
+    Option,
+    '--no-warn-python-deprecation',
+    dest='silence_python_deprecation_warnings',
+    action='store_true',
+    default=False,
+    help='Silence deprecation warns for Python versions.'
```

Comment:
I would phrase it like "Silence deprecation warnings for upcoming unsupported Pythons."